src/index/vocabulary/SplitVocabulary.h

```diff
@@ -0,0 +1,183 @@
+    if (useSpecial) {
+      WordAndIndex subResult = underlyingSpecial_.lower_bound(word, comparator);
+      if (subResult.isEnd()) {
+        return subResult;
+      }
+      return {subResult.word(), subResult.index() | specialVocabMarker};
+    } else {
+      return underlyingMain_.lower_bound(word, comparator);
+    }
+  }
+  template <typename InternalStringType, typename Comparator>
+  WordAndIndex upper_bound(const InternalStringType& word,
+                           Comparator comparator,
+                           bool useSpecial = false) const {
+    if (useSpecial) {
+      WordAndIndex subResult = underlyingSpecial_.upper_bound(word, comparator);
+      if (subResult.isEnd()) {
+        return subResult;
+      }
+      return {subResult.word(), subResult.index() | specialVocabMarker};
+    } else {
+      return underlyingMain_.upper_bound(word, comparator);
+    }
+  }
+
+  MainVocabulary& getUnderlyingMainVocabulary() { return underlyingMain_; }
+
+  const MainVocabulary& getUnderlyingMainVocabulary() const {
+    return underlyingMain_;
+  }
+
+  SpecialVocabulary& getUnderlyingSpecialVocabulary() {
+    return underlyingSpecial_;
+  }
+
+  const SpecialVocabulary& getUnderlyingSpecialVocabulary() const {
+    return underlyingSpecial_;
+  }
+
+  void open(const std::string& filename);
+
+  // This word writer writes words to different vocabularies depending on their
+  // content.
+  using MainWWPtr = std::unique_ptr<typename MainVocabulary::WordWriter>;
+  using SpecialWWPtr = std::unique_ptr<typename SpecialVocabulary::WordWriter>;
+  class WordWriter {
+   private:
+    MainWWPtr underlyingWordWriter_;
+    SpecialWWPtr underlyingSpecialWordWriter_;
+    std::string readableName_ = "";
+
+   public:
+    WordWriter(const MainVocabulary& mainVocabulary,
+               const SpecialVocabulary& specialVocabulary,
+               const std::string& filename);
+
+    // Add the next word to the vocabulary and return its index.
+    uint64_t operator()(std::string_view word, bool isExternal);
+
+    // Finish the writing on both underlying word writers. After this no more
+    // calls to `operator()` are allowed.
+    void finish();
+
+    std::string& readableName() { return readableName_; }
+  };
+
+  using WWPtr = std::unique_ptr<WordWriter>;
+  WWPtr makeWordWriterPtr(const std::string& filename) const {
+    return std::make_unique<WordWriter>(underlyingMain_, underlyingSpecial_,
+                                        filename);
+  }
+};
+
+// Concrete implementations of split function and split filename function
+
+static constexpr std::string_view GEO_LITERAL_SUFFIX =
+    ad_utility::constexprStrCat<"\"^^<", GEO_WKT_LITERAL, ">">();
+
+// Split function
+inline bool geoSplitFunc(std::string_view word) {
+  return word.starts_with("\"") && word.ends_with(GEO_LITERAL_SUFFIX);
+};
+
+// Split filename function
+inline std::array<std::string, 2> geoFilenameFunc(std::string base) {
```

Comment:
Can simply take a `string_view` as argument (then use StrCat for the second argument;.